samcli/commands/init/init_templates.py

```diff
@@ -0,0 +1,150 @@
+"""
+Manages the set of application templates.
+"""
+
+import itertools
+import json
+import os
+import platform
+import shutil
+import subprocess
+
+from pathlib import Path  # must come after Py2.7 deprecation
+
+import click
+
+from samcli.cli.main import global_cfg
+from samcli.commands.exceptions import UserException
+from samcli.local.common.runtime_template import RUNTIME_DEP_TEMPLATE_MAPPING
+
+
+class InitTemplates:
+    def __init__(self, no_interactive=False, auto_clone=True):
+        self._repo_url = "https://github.com/awslabs/aws-sam-cli-app-templates.git"
+        self._repo_name = "aws-sam-cli-app-templates"
+        self.repo_path = None
+        self.clone_attempted = False
+        self._no_interactive = no_interactive
+        self._auto_clone = auto_clone
+
+    def prompt_for_location(self, runtime, dependency_manager):
+        options = self.init_options(runtime, dependency_manager)
+        choices = map(str, range(1, len(options) + 1))
+        choice_num = 1
+        for o in options:
+            if o.get("displayName") is not None:
```

Comment:
is displayname part of cookiecutter options? in the cookiecutter.json file. I see it used in the manifest in a test below. Never mind looks to be part of manifest.json

https://github.com/awslabs/aws-sam-cli-app-templates/blob/master/manifest.json